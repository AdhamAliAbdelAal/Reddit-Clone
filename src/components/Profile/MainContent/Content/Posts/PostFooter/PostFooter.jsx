import ChatBubbleOutlineOutlinedIcon from '@mui/icons-material/ChatBubbleOutlineOutlined';
import ShortcutOutlinedIcon from '@mui/icons-material/ShortcutOutlined';
import CheckCircleOutlineOutlinedIcon from '@mui/icons-material/CheckCircleOutlineOutlined';
import BlockOutlinedIcon from '@mui/icons-material/BlockOutlined';
import AdminPanelSettingsOutlinedIcon from '@mui/icons-material/AdminPanelSettingsOutlined';
import MoreHorizOutlinedIcon from '@mui/icons-material/MoreHorizOutlined';
import CancelPresentationOutlinedIcon from '@mui/icons-material/CancelPresentationOutlined';
import SignalCellularAltOutlinedIcon from '@mui/icons-material/SignalCellularAltOutlined';
import ModeEditOutlinedIcon from '@mui/icons-material/ModeEditOutlined';
import { useState } from 'react';
import { ClickAwayListener, Divider } from '@mui/material';

import BookmarkBorderOutlinedIcon from '@mui/icons-material/BookmarkBorderOutlined';
import PushPinOutlinedIcon from '@mui/icons-material/PushPinOutlined';
import VisibilityOffOutlinedIcon from '@mui/icons-material/VisibilityOffOutlined';
import DeleteOutlineOutlinedIcon from '@mui/icons-material/DeleteOutlineOutlined';
import CropSquareOutlinedIcon from '@mui/icons-material/CropSquareOutlined';
import {
  ElementBox, FooterBox, FooterText, SelectBox, SelectItem,
} from './styles';

function PostFooter(props) {
  const { subTitle, numComments } = props;

  const [showList, setShowList] = useState(false);

  // handle disable the list when click away
  const handleClick = () => {
    setShowList((prev) => !prev);
  };

  const handleClickAway = () => {
    setShowList(false);
  };

  return (
    <FooterBox>
      <ElementBox>
        <ChatBubbleOutlineOutlinedIcon />
        <FooterText variant="caption" responsiveshare={true.toString()}>{numComments}</FooterText>
      </ElementBox>

      <ElementBox>
        <ShortcutOutlinedIcon />
        <FooterText variant="caption" responsiveshare={true.toString()}>Share</FooterText>
      </ElementBox>
      <ElementBox condition2={(subTitle === 'Edited').toString()}>
        <CheckCircleOutlineOutlinedIcon />
        <FooterText variant="caption" responsiveapprove={true.toString()}>Approve</FooterText>
      </ElementBox>
      <ElementBox condition={(subTitle === 'Spam').toString()}>
        <BlockOutlinedIcon />
        <FooterText variant="caption" responsiveapprove={true.toString()}>Removed</FooterText>
      </ElementBox>
      <ElementBox>
        <CancelPresentationOutlinedIcon />
        <FooterText variant="caption" responsive={true.toString()}>Spam</FooterText>
      </ElementBox>
      <ElementBox>
        <AdminPanelSettingsOutlinedIcon />
      </ElementBox>
      <ElementBox responsive={true.toString()}>
        <SignalCellularAltOutlinedIcon sx={{ color: '#b279ff' }} />
        <FooterText variant="caption">Insights</FooterText>
      </ElementBox>
      <ClickAwayListener onClickAway={handleClickAway}>
        <ElementBox>
          <MoreHorizOutlinedIcon onClick={handleClick} />
        </ElementBox>

      </ClickAwayListener>
      {showList && (
      <SelectBox>
        <SelectItem>
          <ModeEditOutlinedIcon sx={{ marginRight: 1 }} />
          Edit Post
        </SelectItem>
        <Divider />
        <SelectItem>
          <BookmarkBorderOutlinedIcon sx={{ marginRight: 1 }} />
          Save
        </SelectItem>
        <Divider />
        <SelectItem>
          <PushPinOutlinedIcon sx={{ marginRight: 1 }} />
          Pin Post To Profile
        </SelectItem>
        <Divider />
        <SelectItem>
          <VisibilityOffOutlinedIcon sx={{ marginRight: 1 }} />
          Hide
        </SelectItem>
        <Divider />
        <SelectItem>
          <DeleteOutlineOutlinedIcon sx={{ marginRight: 1 }} />
          Delete
        </SelectItem>
        <Divider />
        <SelectItem>
          <CropSquareOutlinedIcon sx={{ marginRight: 1 }} />
          Mark As OC
        </SelectItem>
        <Divider />
        <SelectItem>
          <CropSquareOutlinedIcon sx={{ marginRight: 1 }} />
          Mark As Spoiler
        </SelectItem>
        <Divider />
        <SelectItem>
          <CropSquareOutlinedIcon sx={{ marginRight: 1 }} />
          Mark As NSFW
        </SelectItem>
        <Divider />
        <SelectItem>
          <CropSquareOutlinedIcon sx={{ marginRight: 1 }} />
          Send Me Reply Notifications
        </SelectItem>

      </SelectBox>
      )}
    </FooterBox>
  );
}

export default PostFooter;
