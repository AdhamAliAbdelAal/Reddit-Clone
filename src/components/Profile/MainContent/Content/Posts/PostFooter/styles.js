import { Box, Button, Typography } from '@mui/material';
import { styled } from '@mui/material/styles';

export const FooterBox = styled(Box)(() => ({
  width: '100%',
  height: 40,
  display: 'flex',
  alignItems: 'center',

}));
export const FooterText = styled(Typography)(() => ({
  marginLeft: 5,
  display: 'flex',
  alignItems: 'center',
  fontWeight: 700,
}));
export const ElementBox = styled(Box)(({ condition, condition2, theme }) => ({
  height: '100%',
  display: 'flex',
  marginRight: 1,
  alignItems: 'center',
  borderRadius: 4,
  padding: '5px 6px',
  color: theme.palette.grey[500],
  '&:hover': {
    backgroundColor: '#1a1a1b1a',
  },
  ...((condition === 'true') && {
    color: '#ff585b80',
    '&:hover': {
    },
  }),
  ...((condition2 === 'true') && {
    color: '#46d16087',
    '&:hover': {
    },
  }),
}));

export const SelectBox = styled(Box)(() => ({
  width: 260,
  marginTop: 330,
  marginLeft: 500,
  border: ' 1px solid #eee',
  borderRadius: 4,
  boxShadow: '0 2px 4px 0 #eee',
  backgroundColor: 'white',
  position: 'absolute',
  zIndex: 10,

}));

export const SelectItem = styled(Button)(() => ({
  padding: 5,
  color: '#878a8c',
  fontSize: 14,
  fontWeight: 700,
  width: '100%',
  justifyContent: 'left',
  textTransform: 'none',
  // not working
  '& .MuiButtonBase-root:hover': {
    color: 'black',
    backgroundColor: '#d7d7d7',
  },
}));
