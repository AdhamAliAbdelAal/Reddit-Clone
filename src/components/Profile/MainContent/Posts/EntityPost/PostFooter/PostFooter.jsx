import ChatBubbleOutlineOutlinedIcon from '@mui/icons-material/ChatBubbleOutlineOutlined';
import ShortcutOutlinedIcon from '@mui/icons-material/ShortcutOutlined';
import CheckCircleOutlineOutlinedIcon from '@mui/icons-material/CheckCircleOutlineOutlined';
import BlockOutlinedIcon from '@mui/icons-material/BlockOutlined';
import AdminPanelSettingsOutlinedIcon from '@mui/icons-material/AdminPanelSettingsOutlined';
import MoreHorizOutlinedIcon from '@mui/icons-material/MoreHorizOutlined';
import CancelPresentationOutlinedIcon from '@mui/icons-material/CancelPresentationOutlined';
import SignalCellularAltOutlinedIcon from '@mui/icons-material/SignalCellularAltOutlined';
import ModeEditOutlinedIcon from '@mui/icons-material/ModeEditOutlined';
import { useState } from 'react';
import { ClickAwayListener, Divider } from '@mui/material';

import BookmarkBorderOutlinedIcon from '@mui/icons-material/BookmarkBorderOutlined';
import PushPinOutlinedIcon from '@mui/icons-material/PushPinOutlined';
import VisibilityOffOutlinedIcon from '@mui/icons-material/VisibilityOffOutlined';
import DeleteOutlineOutlinedIcon from '@mui/icons-material/DeleteOutlineOutlined';
import CropSquareOutlinedIcon from '@mui/icons-material/CropSquareOutlined';
import UnfoldLessOutlinedIcon from '@mui/icons-material/UnfoldLessOutlined';
import UnfoldMoreOutlinedIcon from '@mui/icons-material/UnfoldMoreOutlined';
import {
  ElementBox, FooterBox, FooterText, SelectBox, SelectItem,
} from './styles';

function PostFooter(props) {
  const numComments = 0;
  const {
    subTitle, handleExpand, expand, upvoted,
  } = props;

  const [showList, setShowList] = useState(false);
  // handle disable the list when click away
  const handleClick = () => {
    setShowList((prev) => !prev);
  };

  const handleClickAway = () => {
    setShowList(false);
  };

  return (
    <FooterBox>
      <ElementBox>
        {expand ? <UnfoldLessOutlinedIcon sx={{ rotate: '-45deg' }} onClick={() => { handleExpand(); }} />
          : <UnfoldMoreOutlinedIcon sx={{ rotate: '-45deg' }} onClick={() => { handleExpand(); }} />}
      </ElementBox>
      <Divider orientation="vertical" variant="middle" flexItem />
      <ElementBox>
        <ChatBubbleOutlineOutlinedIcon />
        <FooterText variant="caption">{numComments}</FooterText>
      </ElementBox>

      <ElementBox>
        <ShortcutOutlinedIcon />
        <FooterText variant="caption">Share</FooterText>
      </ElementBox>
      <ElementBox condition2={(subTitle === 'Edited').toString()}>
        <CheckCircleOutlineOutlinedIcon />
        <FooterText variant="caption">Approve</FooterText>
      </ElementBox>
      <ElementBox condition={(subTitle === 'Spam').toString()}>
        <BlockOutlinedIcon />
        <FooterText variant="caption">Removed</FooterText>
      </ElementBox>
      <ElementBox>
        <CancelPresentationOutlinedIcon />
        <FooterText variant="caption">Spam</FooterText>
      </ElementBox>
      <ElementBox>
        <AdminPanelSettingsOutlinedIcon />
      </ElementBox>
      {
      !upvoted && (
      <ElementBox>
        <SignalCellularAltOutlinedIcon sx={{ color: '#b279ff' }} />
        <FooterText variant="caption">Insights</FooterText>
      </ElementBox>
      )
      }
      <ClickAwayListener onClickAway={handleClickAway}>
        <ElementBox>
          <MoreHorizOutlinedIcon onClick={handleClick} />
        </ElementBox>

      </ClickAwayListener>
      {showList && (
      <SelectBox>
        <SelectItem>
          <ModeEditOutlinedIcon sx={{ marginRight: 1 }} />
          Edit Post
        </SelectItem>
        <Divider />
        <SelectItem>
          <BookmarkBorderOutlinedIcon sx={{ marginRight: 1 }} />
          Save
        </SelectItem>
        <Divider />
        <SelectItem>
          <PushPinOutlinedIcon sx={{ marginRight: 1 }} />
          Pin Post To Profile
        </SelectItem>
        <Divider />
        <SelectItem>
          <VisibilityOffOutlinedIcon sx={{ marginRight: 1 }} />
          Hide
        </SelectItem>
        <Divider />
        <SelectItem>
          <DeleteOutlineOutlinedIcon sx={{ marginRight: 1 }} />
          Delete
        </SelectItem>
        <Divider />
        <SelectItem>
          <CropSquareOutlinedIcon sx={{ marginRight: 1 }} />
          Mark As OC
        </SelectItem>
        <Divider />
        <SelectItem>
          <CropSquareOutlinedIcon sx={{ marginRight: 1 }} />
          Mark As Spoiler
        </SelectItem>
        <Divider />
        <SelectItem>
          <CropSquareOutlinedIcon sx={{ marginRight: 1 }} />
          Mark As NSFW
        </SelectItem>
        <Divider />
        <SelectItem>
          <CropSquareOutlinedIcon sx={{ marginRight: 1 }} />
          Send Me Reply Notifications
        </SelectItem>

      </SelectBox>
      )}
    </FooterBox>
  );
}

export default PostFooter;
