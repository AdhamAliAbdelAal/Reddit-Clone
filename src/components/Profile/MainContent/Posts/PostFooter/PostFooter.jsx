import ChatBubbleOutlineOutlinedIcon from '@mui/icons-material/ChatBubbleOutlineOutlined';
import ShortcutOutlinedIcon from '@mui/icons-material/ShortcutOutlined';
import CheckCircleOutlineOutlinedIcon from '@mui/icons-material/CheckCircleOutlineOutlined';
import BlockOutlinedIcon from '@mui/icons-material/BlockOutlined';
import AdminPanelSettingsOutlinedIcon from '@mui/icons-material/AdminPanelSettingsOutlined';
import MoreHorizOutlinedIcon from '@mui/icons-material/MoreHorizOutlined';
import CancelPresentationOutlinedIcon from '@mui/icons-material/CancelPresentationOutlined';
import SignalCellularAltOutlinedIcon from '@mui/icons-material/SignalCellularAltOutlined';
import ModeEditOutlinedIcon from '@mui/icons-material/ModeEditOutlined';
import { useState } from 'react';
import { ClickAwayListener, Divider } from '@mui/material';

import BookmarkBorderOutlinedIcon from '@mui/icons-material/BookmarkBorderOutlined';
import VisibilityOffOutlinedIcon from '@mui/icons-material/VisibilityOffOutlined';
import UnfoldLessOutlinedIcon from '@mui/icons-material/UnfoldLessOutlined';
import UnfoldMoreOutlinedIcon from '@mui/icons-material/UnfoldMoreOutlined';
import FlagOutlinedIcon from '@mui/icons-material/FlagOutlined';
import BookmarksOutlinedIcon from '@mui/icons-material/BookmarksOutlined';
import VisibilityOffIcon from '@mui/icons-material/VisibilityOff';
import {
  ElementBox, FooterBox, FooterText,
} from './styles';
import PostFooterList from './PostFooterList/PostFooterList';

/** Footer of the post that contain all icons
 * @return {React.Component} - PostFooter
 * @param {function} handleExpand - pass the event to the parent to show the paragraph of the post
 * @param {state} expand - pass a state to the parent to change the icon
 * @param {object} entity - contains all info of the post
 */
function PostFooter(props) {
  const {
    numComments,
    approved,
    removed,
    spam,
    handleExpand,
    expand,
    voted,
    postedByOthers,
    saved,
    hidden,
  } = props;
  const [isHidden, setIsHidden] = useState(hidden);
  const [isSaved, setIsSaved] = useState(saved);
  const [showList, setShowList] = useState(false);
  // handle disable the list when click away
  const handleClick = () => {
    setShowList((prev) => !prev);
  };

  const handleClickAway = () => {
    setShowList(false);
  };

  // switch icon to hidden post and vice verse
  const handleClickHide = () => {
    setIsHidden((prev) => !prev);
  };

  // switch icon to saved post and vice versa
  const handleClickSave = () => {
    setIsSaved((prev) => !prev);
  };

  const moderator = [
    {
      id: 1, text: 'Approved', condition: approved, icon: <CheckCircleOutlineOutlinedIcon />,
    },
    {
      id: 2, text: 'Removed', condition: removed, icon: <BlockOutlinedIcon />,
    },
    {
      id: 3, text: 'Spam', condition: spam, icon: <CancelPresentationOutlinedIcon />,
    },
    {
      id: 4, text: '', condition: false, icon: <AdminPanelSettingsOutlinedIcon />,
    },
  ];
  const publisher = [
    {
      id: 1, text: 'Edit Post', icon: <ModeEditOutlinedIcon />,
    },
    isSaved ? {
      id: 2, text: 'Unsave', icon: <BookmarksOutlinedIcon />, func: true,
    } : {
      id: 2, text: 'Save', icon: <BookmarkBorderOutlinedIcon />, func: true,
    },
    isHidden ? {
      id: 3, text: 'Unhide', icon: <VisibilityOffIcon />, func: false,
    } : {
      id: 3, text: 'Hide', icon: <VisibilityOffOutlinedIcon />, func: false,
    },
  ];

  return (
    <FooterBox>
      <ElementBox>
        {expand ? <UnfoldLessOutlinedIcon sx={{ rotate: '-45deg' }} onClick={() => { handleExpand(); }} />
          : <UnfoldMoreOutlinedIcon sx={{ rotate: '-45deg' }} onClick={() => { handleExpand(); }} />}
      </ElementBox>
      <Divider orientation="vertical" variant="middle" flexItem />
      <ElementBox>
        <ChatBubbleOutlineOutlinedIcon />
        <FooterText variant="caption" responsiveshare={true.toString()}>{numComments}</FooterText>
      </ElementBox>
      <ElementBox>
        <ShortcutOutlinedIcon />
        <FooterText variant="caption" responsiveshare={true.toString()}>Share</FooterText>
      </ElementBox>
      {(!voted) && moderator.map((entity) => (
        <ElementBox
          key={entity.id}
          approved={(entity.condition && entity.text === 'Approved')?.toString()}
          spam={(entity.condition && (entity.text === 'Spam' || entity.text === 'Removed'))?.toString()}
        >
          {entity.icon}
          <FooterText variant="caption" condition={true.toString()}>{entity.text}</FooterText>
        </ElementBox>
      ))}
      {(!voted) && (
      <ElementBox condition2={true.toString()}>
        <SignalCellularAltOutlinedIcon sx={{ color: '#b279ff' }} />
        <FooterText variant="caption">Insights</FooterText>
      </ElementBox>
      )}

      {(voted) && publisher.map((entity) => (
        <ElementBox
          key={entity.id}
          onClick={entity.func ? () => { handleClickSave(); } : () => { handleClickHide(); }}
          data-testid={(entity.id === 3) && 'hidden'}
        >
          {entity.icon}
          <FooterText variant="caption" condition={true.toString()} data-testid={(entity.id === 3) && 'text-hide'}>{entity.text}</FooterText>
        </ElementBox>
      ))}

      {postedByOthers ? (
        <ElementBox>
          <FlagOutlinedIcon />
          <FooterText variant="caption" condition={true.toString()}>Report</FooterText>
        </ElementBox>
      ) : (
        <ClickAwayListener onClickAway={handleClickAway}>
          <ElementBox>
            <MoreHorizOutlinedIcon onClick={handleClick} />
            {showList && (
              <PostFooterList />
            )}
          </ElementBox>
        </ClickAwayListener>
      )}

    </FooterBox>
  );
}

export default PostFooter;
