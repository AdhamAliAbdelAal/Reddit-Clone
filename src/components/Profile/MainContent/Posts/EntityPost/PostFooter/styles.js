import { Box, Button, Typography } from '@mui/material';
import { styled } from '@mui/material/styles';

export const FooterBox = styled(Box)(() => ({
  width: '100%',
  height: 36,
  display: 'flex',
  alignItems: 'center',

}));
export const FooterText = styled(Typography)(({ theme, condition, responsiveshare }) => ({
  marginLeft: 5,
  display: 'flex',
  alignItems: 'center',
  fontWeight: 700,
  ...((condition === 'true') && {
    [theme.breakpoints.between('0', '1160')]: {
      display: 'none',
    },
  }),
  ...((responsiveshare === 'true') && {
    [theme.breakpoints.between('0', '500')]: {
      display: 'none',
    },
  }),

}));
export const ElementBox = styled(Box)(({
  spam, approved, theme, condition2,
}) => ({
  height: '100%',
  display: 'flex',
  marginRight: 1,
  alignItems: 'center',
  borderRadius: 4,
  padding: '5px 6px',
  color: theme.palette.grey[500],
  '&:hover': {
    backgroundColor: '#1a1a1b1a',
  },
  ...((spam === 'true') && {
    color: '#ff585b80',
    '&:hover': {
    },
  }),
  ...((approved === 'true') && {
    color: '#46d16087',
    '&:hover': {
    },
  }),
  ...((condition2) && {
    [theme.breakpoints.between('0', '1050')]: {
      display: 'none',
    },
  }),
}));

export const SelectBox = styled(Box)(() => ({
  width: 260,
  marginTop: 330,
  border: ' 1px solid #eee',
  borderRadius: 4,
  boxShadow: '0 2px 4px 0 #eee',
  backgroundColor: 'white',
  position: 'absolute',
  zIndex: 10,

}));

export const SelectItem = styled(Button)(() => ({
  padding: 5,
  color: '#878a8c',
  fontSize: 14,
  fontWeight: 700,
  width: '100%',
  justifyContent: 'left',
  textTransform: 'none',
  // not working
  '& .MuiButtonBase-root:hover': {
    color: 'black',
    backgroundColor: '#d7d7d7',
  },
}));
