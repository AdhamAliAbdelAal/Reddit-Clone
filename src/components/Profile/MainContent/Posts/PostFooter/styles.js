import { Box, Typography } from '@mui/material';
import { styled } from '@mui/material/styles';

export const FooterBox = styled(Box)(() => ({
  width: '100%',
  height: 36,
  display: 'flex',
  alignItems: 'center',

}));
export const FooterText = styled(Typography)(({ theme, condition, responsiveshare }) => ({
  marginLeft: 5,
  display: 'flex',
  alignItems: 'center',
  fontWeight: 700,
  ...((condition === 'true') && {
    [theme.breakpoints.between('0', '1160')]: {
      display: 'none',
    },
  }),
  ...((responsiveshare === 'true') && {
    [theme.breakpoints.between('0', '500')]: {
      display: 'none',
    },
  }),

}));
export const ElementBox = styled(Box)(({
  spam, approved, theme, condition2, condition, show,
}) => ({
  height: '100%',
  display: 'flex',
  marginRight: 1,
  alignItems: 'center',
  borderRadius: 4,
  padding: '5px 6px',
  color: theme.palette.grey[500],
  '&:hover': {
    backgroundColor: '#1a1a1b1a',
  },
  ...((spam === 'true') && {
    color: '#ff585b80',
    '&:hover': {
    },
  }),
  ...((approved === 'true') && {
    color: '#46d16087',
    '&:hover': {
    },
  }),
  ...((condition2) && {
    [theme.breakpoints.between('0', '1050')]: {
      display: 'none',
    },
  }),
  ...((condition) && {
    [theme.breakpoints.between('0', '450')]: {
      display: 'none',
    },
  }),
  ...((show === 'true') && {
    display: 'none',
  }),
  ...((show === 'true') && {
    [theme.breakpoints.between('0', '450')]: {
      display: 'block',
    },
  }),
}));
