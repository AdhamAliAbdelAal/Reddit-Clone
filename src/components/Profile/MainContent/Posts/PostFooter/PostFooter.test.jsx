import '@testing-library/jest-dom';
// import renderer from 'react-test-renderer';
// import {
//   fireEvent,
//   render, screen,
// } from '@testing-library/react';
import renderer from 'react-test-renderer';
import PostFooter from './PostFooter';

// it('open posts tap', async () => {
//   // window.history.pushState({}, '', '/Login');
//   render(<PostFooter />);
//   const btn = screen.getByTestId('hidden');
//   fireEvent.click(btn);
//   const cancelHidden = screen.getByTestId('text-hide');
//   expect(cancelHidden.innerHTML).toEqual('Unhide');

// // expect(cancelHidden).toBeInTheDocument();
// //   expect(global.window.location.pathname).toContain('/submitted');
// });

test('test snapshot', async () => {
  const tree = renderer.create(<PostFooter />).toJSON();
  expect(tree).toMatchSnapshot();
});
