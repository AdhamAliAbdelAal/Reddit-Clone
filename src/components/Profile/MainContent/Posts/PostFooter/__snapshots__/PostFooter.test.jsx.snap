// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div
  className="MuiBox-root css-146v974"
>
  <div
    className="MuiBox-root css-xv0eud"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1m9p8mk-MuiSvgIcon-root"
      data-testid="ForwardOutlinedIcon"
      focusable="false"
      onClick={[Function]}
      viewBox="0 0 24 24"
    >
      <path
        d="M14 8.83 17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
      />
    </svg>
    <span
      className="MuiTypography-root MuiTypography-caption css-gj0u27-MuiTypography-root"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-hshnfv-MuiSvgIcon-root"
      data-testid="ForwardOutlinedIcon"
      focusable="false"
      onClick={[Function]}
      viewBox="0 0 24 24"
    >
      <path
        d="M14 8.83 17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
      />
    </svg>
  </div>
  <div
    className="MuiBox-root css-orlaaz"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1or9raf-MuiSvgIcon-root"
      data-testid="UnfoldMoreOutlinedIcon"
      focusable="false"
      onClick={[Function]}
      viewBox="0 0 24 24"
    >
      <path
        d="M12 5.83 15.17 9l1.41-1.41L12 3 7.41 7.59 8.83 9 12 5.83zm0 12.34L8.83 15l-1.41 1.41L12 21l4.59-4.59L15.17 15 12 18.17z"
      />
    </svg>
  </div>
  <hr
    className="MuiDivider-root MuiDivider-middle MuiDivider-vertical MuiDivider-flexItem css-16bghm4-MuiDivider-root"
  />
  <div
    className="MuiBox-root css-orlaaz"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="ChatBubbleOutlineOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 2H4c-1.1 0-2 .9-2 2v18l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm0 14H6l-2 2V4h16v12z"
      />
    </svg>
    <span
      className="MuiTypography-root MuiTypography-caption css-1w4fbeu-MuiTypography-root"
      responsiveshare="true"
    >
       Comments
    </span>
  </div>
  <div
    className="MuiBox-root css-orlaaz"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="ShortcutOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m15 5-1.41 1.41L15 7.83 17.17 10H8c-2.76 0-5 2.24-5 5v4h2v-4c0-1.65 1.35-3 3-3h9.17L15 14.17l-1.41 1.41L15 17l6-6-6-6z"
      />
    </svg>
    <span
      className="MuiTypography-root MuiTypography-caption css-1w4fbeu-MuiTypography-root"
      responsiveshare="true"
    >
      Share
    </span>
  </div>
  <div
    awardedit="true"
    className="MuiBox-root css-1bkcqq4"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="ModeEditOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M3 21h3.75L17.81 9.94l-3.75-3.75L3 17.25V21zm2-2.92 9.06-9.06.92.92L5.92 19H5v-.92zM18.37 3.29a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83c.39-.39.39-1.02 0-1.41l-2.34-2.34z"
      />
    </svg>
    <span
      className="MuiTypography-root MuiTypography-caption css-1w4fbeu-MuiTypography-root"
      responsiveshare="true"
    >
      Edit Post
    </span>
  </div>
  <div
    className="MuiBox-root css-196x17m"
    condition="true"
    data-testid={false}
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="BookmarkBorderOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M17 3H7c-1.1 0-2 .9-2 2v16l7-3 7 3V5c0-1.1-.9-2-2-2zm0 15-5-2.18L7 18V5h10v13z"
      />
    </svg>
    <span
      className="MuiTypography-root MuiTypography-caption css-1a0mhoe-MuiTypography-root"
      condition="true"
      data-testid={false}
    >
      Save
    </span>
  </div>
  <div
    className="MuiBox-root css-196x17m"
    condition="true"
    data-testid="hidden"
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="VisibilityOffOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 6c3.79 0 7.17 2.13 8.82 5.5-.59 1.22-1.42 2.27-2.41 3.12l1.41 1.41c1.39-1.23 2.49-2.77 3.18-4.53C21.27 7.11 17 4 12 4c-1.27 0-2.49.2-3.64.57l1.65 1.65C10.66 6.09 11.32 6 12 6zm-1.07 1.14L13 9.21c.57.25 1.03.71 1.28 1.28l2.07 2.07c.08-.34.14-.7.14-1.07C16.5 9.01 14.48 7 12 7c-.37 0-.72.05-1.07.14zM2.01 3.87l2.68 2.68C3.06 7.83 1.77 9.53 1 11.5 2.73 15.89 7 19 12 19c1.52 0 2.98-.29 4.32-.82l3.42 3.42 1.41-1.41L3.42 2.45 2.01 3.87zm7.5 7.5 2.61 2.61c-.04.01-.08.02-.12.02-1.38 0-2.5-1.12-2.5-2.5 0-.05.01-.08.01-.13zm-3.4-3.4 1.75 1.75c-.23.55-.36 1.15-.36 1.78 0 2.48 2.02 4.5 4.5 4.5.63 0 1.23-.13 1.77-.36l.98.98c-.88.24-1.8.38-2.75.38-3.79 0-7.17-2.13-8.82-5.5.7-1.43 1.72-2.61 2.93-3.53z"
      />
    </svg>
    <span
      className="MuiTypography-root MuiTypography-caption css-1a0mhoe-MuiTypography-root"
      condition="true"
      data-testid="text-hide"
    >
      Hide
    </span>
  </div>
  <div
    className="MuiBox-root css-orlaaz"
    onClick={[Function]}
    onTouchEnd={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="MoreHorizOutlinedIcon"
      focusable="false"
      onClick={[Function]}
      viewBox="0 0 24 24"
    >
      <path
        d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
      />
    </svg>
  </div>
</div>
`;
